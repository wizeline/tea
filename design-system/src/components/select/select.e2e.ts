import { newE2EPage } from '@stencil/core/testing';

describe('wz-select', () => {
  it('should render a select-component', async () => {
    const page = await newE2EPage();
    await page.setContent(`
      <wz-select>
        <option selected value="1">First Option</option>
        <option option="2">Second Option</option>
      </wz-select>`);
    const select = await page.find('wz-select');
    expect(select.outerHTML).toMatchSnapshot();
  });
});
