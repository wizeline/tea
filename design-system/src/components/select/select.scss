wz-select {
  position: relative;
  select {
    appearance: none;
    height: 40px;
    padding-left: 12px;
    padding-right: 28px;
    border-radius: 3px;
    border: solid 1px var(--ink-light);
    background-color: var(--surface-base);
    font-size: 14px;
  }
  &:after {
    content: '';
    position: absolute;
    top: 7px;
    right: 8px;
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 0 4px 4px;
    border-color: transparent transparent var(--ink-base) transparent;
    transform: rotateX(180deg);
    pointer-events: none;
  }
}
