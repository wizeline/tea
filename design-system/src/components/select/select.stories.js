import { storiesOf } from '@storybook/html';

storiesOf('Select', module).add(
  'default',
  () => `
  <wz-select onchange="(function(e) { console.log(this.event.target.value) })()">
    <option value="volvo">Volvo</option>
    <option value="saab" selected>Saab</option>
    <option value="mercedes">Mercedes</option>
    <option value="audi">Audi</option>
  </wz-select>`,
);
