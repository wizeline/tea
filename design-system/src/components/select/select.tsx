import { Component } from '@stencil/core';

@Component({
  tag: 'wz-select',
  styleUrl: 'select.scss',
  shadow: false,
})
export class Select {
  render() {
    return (
      <select>
        <slot />
      </select>
    );
  }
}
