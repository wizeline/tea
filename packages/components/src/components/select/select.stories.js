import { storiesOf } from '@storybook/html';

storiesOf('Select', module).add(
  'default',
  () => `
  <script>
  function change() {
    console.log('click');
  }
  </script>
  <wz-select onchange="(function(e) { console.log(this.event.target.value) })()">
    <option value="volvo">Volvo</option>
    <option value="saab" selected>Saab</option>
    <option value="mercedes">Mercedes</option>
    <option value="audi">Audi</option>
  </wz-select>`,
);
