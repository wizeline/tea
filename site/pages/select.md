### Select
```html
showSource: true
---
<wz-select>
  <option value="volvo">Volvo</option>
  <option selected value="saab">Saab</option>
  <option value="mercedes">Mercedes</option>
  <option value="audi">Audi</option>
</wz-select>
```
