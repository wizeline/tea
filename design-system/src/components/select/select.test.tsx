import { Select } from './select';
import { TestWindow } from '@stencil/core/testing';

describe('wz-select', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [Select],
      html: `
      <wz-select>
        <option selected value="1">First Option</option>
        <option option="2">Second Option</option>
      </wz-select>`,
    });

    expect(element).toMatchSnapshot();
  });
});
