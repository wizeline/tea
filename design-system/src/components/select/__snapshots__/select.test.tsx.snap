// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-select renders value 1`] = `
<wz-select
  class="hydrated"
>
  
  
  
  
  
  
  <select>
    
    
        
    <option
      selected=""
      value="1"
    >
      First Option
    </option>
    
        
    <option
      option="2"
    >
      Second Option
    </option>
    
      
  </select>
</wz-select>
`;
